sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -632,9 +632,51 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     val rawTable = getRawTable(db, table)
     val withNewSchema = rawTable.copy(schema = schema)
     verifyColumnNames(withNewSchema)
+
+    val newProperties = collection.mutable.Map[String, String]()
+    for ((property, value) <- withNewSchema.properties) {
+      newProperties.put(property, value)
+    }
+    for ((property, value) <- tableMetaToTableProps(withNewSchema)) {
+      newProperties.put(property, value)
+    }
+
+    // Since the table schema will be changed, check if the bucketing and sort columns are still
```

Comment:
I don't think we need to do this, according to the document of `ExternalCatalog.alterTableSchema`, the caller will guarantee that the new schema still contains partition columns and bucket columns.